auction-house/program/src/execute_sale/mod.rs

```diff
@@ -0,0 +1,907 @@
+    #[account(mut, seeds=[PREFIX.as_bytes(), seller.key().as_ref(), auction_house.key().as_ref(), token_account.key().as_ref(), auction_house.treasury_mint.as_ref(), token_mint.key().as_ref(), &u64::MAX.to_le_bytes(), &token_size.to_le_bytes()], bump=seller_trade_state.to_account_info().data.borrow()[0])]
+    pub seller_trade_state: UncheckedAccount<'info>,
+
+    /// CHECK: Not dangerous. Account seeds checked in constraint.
+    /// Free seller trade state PDA account encoding a free sell order.
+    #[account(mut, seeds=[PREFIX.as_bytes(), seller.key().as_ref(), auction_house.key().as_ref(), token_account.key().as_ref(), auction_house.treasury_mint.as_ref(), token_mint.key().as_ref(), &0u64.to_le_bytes(), &token_size.to_le_bytes()], bump=free_trade_state_bump)]
+    pub free_trade_state: UncheckedAccount<'info>,
+
+    /// CHECK: Verified through CPI
+    /// The auctioneer program PDA running this auction.
+    pub auctioneer_authority: UncheckedAccount<'info>,
+
+    /// CHECK: Not dangerous. Account seeds checked in constraint.
+    /// The auctioneer PDA owned by Auction House storing scopes.
+    #[account(seeds = [AUCTIONEER.as_bytes(), auction_house.key().as_ref(), auctioneer_authority.key().as_ref()], bump = auction_house.auctioneer_pda_bump)]
+    pub ah_auctioneer_pda: UncheckedAccount<'info>,
+
+    pub token_program: Program<'info, Token>,
+    pub system_program: Program<'info, System>,
+    pub ata_program: Program<'info, AssociatedToken>,
+
+    /// CHECK: Not dangerous. Account seeds checked in constraint.
+    #[account(seeds=[PREFIX.as_bytes(), SIGNER.as_bytes()], bump=program_as_signer_bump)]
+    pub program_as_signer: UncheckedAccount<'info>,
+
+    pub rent: Sysvar<'info, Rent>,
+}
+
+pub fn execute_sale_with_auctioneer<'info>(
+    ctx: Context<'_, '_, '_, 'info, ExecuteSaleWithAuctioneer<'info>>,
+    escrow_payment_bump: u8,
+    free_trade_state_bump: u8,
+    program_as_signer_bump: u8,
+    buyer_price: u64,
+    token_size: u64,
+) -> Result<()> {
+    let auction_house = &ctx.accounts.auction_house;
+    let auctioneer_authority = &ctx.accounts.auctioneer_authority;
+    let ah_auctioneer_pda = &ctx.accounts.ah_auctioneer_pda;
+
+    if !auction_house.has_auctioneer {
+        return Err(AuctionHouseError::NoAuctioneerProgramSet.into());
+    }
+
+    assert_valid_auctioneer_and_scope(
+        &auction_house.key(),
+        &auctioneer_authority.key(),
+        ah_auctioneer_pda,
+        AuthorityScope::ExecuteSale,
+    )?;
+
+    // Duplicate the logic methods to avoid going over the compute limit.
+    execute_auction_sale_logic(
+        ctx,
+        escrow_payment_bump,
+        free_trade_state_bump,
+        program_as_signer_bump,
+        buyer_price,
+        token_size,
+    )
+}
+
+/// Execute sale between provided buyer and seller trade state accounts transferring funds to seller wallet and token to buyer wallet.
+#[inline(never)]
+fn execute_auction_sale_logic<'info>(
```

Comment:
`execute_auctioneer_sale_logic`

I know you're adding `Timed Auctions` with this change. Does it make sense to keep the "auction" wording here? Or does "auctioneer" make more sense. Your call.